lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/summary/Explanation.java

```diff
@@ -19,12 +21,46 @@ public Explanation(List<AttributeSet> resultList,
                        long numInliers,
                        long numOutliers,
                        long creationTimeMs) {
-        itemsets = resultList;
+        itemsets = new ArrayList<>(resultList);
+        itemsets.sort((AttributeSet a, AttributeSet b) -> -a.compareTo(b));
         this.numInliers = numInliers;
         this.numOutliers = numOutliers;
         this.creationTimeMs = creationTimeMs;
     }
 
+    /**
+     * Removes redundant explanations
+     * @return New explanation with redundant itemsets removed.
+     */
+    public Explanation prune() {
+        List<AttributeSet> newItemsets = new ArrayList<>();
+        int n = itemsets.size();
+        for (int i = 0; i < n; i++) {
+            AttributeSet aSet = itemsets.get(i);
+            boolean redundant = false;
+            // an explanation is redundant if it has lower risk ratio (occurs after since sorted)
```

Comment:
Yeah, I figured for convenience we can sort the explanation in the constructor and always have it sorted.